src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/DataGridCell.cs

```diff
@@ -964,28 +964,35 @@ protected override void OnPreviewKeyDown(KeyEventArgs e)
         /// </summary>
         protected override void OnKeyDown(KeyEventArgs e)
         {
-            if ((Keyboard.Modifiers & ModifierKeys.Alt) == ModifierKeys.Alt && (e.Key == Key.Left || e.Key == Key.Right))
+            const ModifierKeys ModifierMask = ModifierKeys.Alt | ModifierKeys.Control | ModifierKeys.Shift | ModifierKeys.Windows;
+            ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;
+
+            if ((modifierKeys & ModifierKeys.Alt) == ModifierKeys.Alt && (e.Key == Key.Left || e.Key == Key.Right))
```

Comment:
I do not think it is right. Such as the `Keyboard.Modifiers` is `ModifierKeys.Alt | ModifierKeys.Control`.

And after the `ModifierKeys modifierKeys = Keyboard.Modifiers & ModifierMask;`, the `modifierKeys` will be `ModifierKeys.Alt | ModifierKeys.Control`.

And `(modifierKeys & ModifierKeys.Alt)` will be `ModifierKeys.Alt`.

Reply:
`(modifierKeys & ModifierKeys.Alt)` = `(Keyboard.Modifiers & ModifierMask) & ModifierKeys.Alt`

= `Keyboard.Modifiers & ModifierMask & ModifierKeys.Alt`

= `Keyboard.Modifiers & (ModifierKeys.Alt | ModifierKeys.Control | ModifierKeys.Shift | ModifierKeys.Windows) & ModifierKeys.Alt`

= `Keyboard.Modifiers & ( (ModifierKeys.Alt | ModifierKeys.Control | ModifierKeys.Shift | ModifierKeys.Windows) & ModifierKeys.Alt)`

= `Keyboard.Modifiers & ModifierKeys.Alt`